### DISCUSSION TITLE
v3.4.15: Major update

### DISCUSSION THREAD

**Discussion by joyfullservice** [joyfullservice/msaccess-vcs-addin#252] [Releases]
Summary
---------------------
This has been a long time in coming, but today we are releasing the **3.4.x version** of the add-in! Containing over 100 code commits, this version includes numerous bug fixes and enhancements. Some of the most notable changes include the transition to VBE format for code module exports, the ability to build a database containing non-compiling code, reduced change "noise" between builds, improved International language support, formatted XML, multiple embedded theme support, and export of extended table properties.

I won't attempt to describe each change in detail, but below are some of the highlights:

Enhancements
---------------------
* [x] Class modules are now **exported using the `.cls` file extension**. This is more consistent with the native VBE export, and provides an easy way to differentiate between standard and class modules in the source files. Legacy class modules are still imported correctly since the module type is determined from the content rather than the extension.
* [x] Module files are now **exported and imported using VBE** instead of `SaveAsText/LoadFromText`. This allows us to export a more universal file format that you can drag and drop into a VBA project, or drag and drop from a VBA project to the file system. Additionally, this now support hidden VBE properties (such as default members or descriptions) not visible in the VBA source code.
* [x] **Non-compiling code** can now be used when building a database from source. This is important when doing collaborative work where the newly built database may require additional code changes to be able to compile. (In the past this would throw an error or fail to complete the build.)
* [x] **Dynamic theme colors** are now sanitized in the source files with multiple levels available in options. This resolves a long-standing issue where builds on different computers would produce slightly different exports. Special thanks to @hecon5 and @A9G-Data-Droid for their help in building and testing this important enhancement!
* [x] A complete list of **codepage mapping** is now included in the add-in, so if you use an International codepage to allow non-english comments in source code, the comments are now correctly translated to UTF-8 for version control, and back to the local code page for import. Thanks, @tst32 for helping make this happen!
* [x] XML data is now **formatted with indentations**, making it much easier to read and compare.
* [x] With the improved XML sanitization function, table definition exports now include **extended properties** (such as description text), allowing these properties to be used when rebuilding from source.
* [x] MSysResources table is now created using **native Access functions** instead of the less predictable temporary form object.
* [x] Added the ability to cancel an import or export operation and still write the log file.

Bug Fixes
---------------------
* [x] Fixed an issue with custom themes not being added, particularly if multiple themes were involved. #240
* [x] Newline characters are now properly substituted when exporting table data as TDF. #251
* [x] Eliminated save-as prompt when importing non-compiling module objects. #236
* [x] Fixed build errors with VBE Project properties used by RubberDuck. #226 
* [x] Resolved export path issue. Thanks @Tanarri! #214 

Other Changes
---------------------
* _**PLEASE NOTE:**_ Due to some changes in export file formats, you may see a large number of changed files the first time you run an Export after upgrading VCS. Note that VBA classes will use the `.cls` file extension going forward. (Git should automatically handle those as renames)

<hr /><em>This discussion was created from the release <a href='https://github.com/joyfullservice/msaccess-vcs-integration/releases/tag/v3.4.15'>v3.4.15: Major update</a>.</em>

**Comment by jlmarquezice** [comment]
Hello, I tried this version today.
I see this in the console, just while: 

"Importing vbe references"
ERROR: Unhandled error, likely before On Error directive
